beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/messaging/MessageEventToPathEventMapper.java

```diff
@@ -17,84 +17,141 @@
 
 import java.time.Duration;
 import java.time.LocalDateTime;
+import java.util.*;
 import java.time.format.DateTimeParseException;
-import java.util.Optional;
 
+import com.expediagroup.beekeeper.core.model.*;
+import com.expediagroup.beekeeper.scheduler.apiary.model.ApiaryLifeCycleEvent;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
-import com.expedia.apiary.extensions.receiver.common.event.AlterPartitionEvent;
-import com.expedia.apiary.extensions.receiver.common.event.AlterTableEvent;
-import com.expedia.apiary.extensions.receiver.common.event.DropPartitionEvent;
-import com.expedia.apiary.extensions.receiver.common.event.DropTableEvent;
-import com.expedia.apiary.extensions.receiver.common.event.EventType;
 import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
 import com.expedia.apiary.extensions.receiver.common.messaging.MessageEvent;
+import com.expediagroup.beekeeper.scheduler.apiary.model.PathEvents;
 
-import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
-import com.expediagroup.beekeeper.core.model.PathStatus;
-import com.expediagroup.beekeeper.scheduler.apiary.model.PathEvent;
+import javax.annotation.PostConstruct;
+
+import static com.expediagroup.beekeeper.core.model.LifeCycleEventType.EXPIRED;
+import static com.expediagroup.beekeeper.core.model.LifeCycleEventType.UNREFERENCED;
 
 @Component
 public class MessageEventToPathEventMapper {
```

Comment:
It looks this would couple mapping/extracting the two event types together, and make it difficult to alter this. So I think probably this class could be split in to two, both which extend a `Mapper` interface -> `OrphanedEventMapper` + `ExpiredEventMapper` `implements MessageEventToPathMapper`
* `MessageReaderAdapter` can then loop through the mappers and add the housekeeping paths to the `BeekeeperEvent`, and at the end add the `MessageEvent`, so it actually becomes more of an adapter, and not just a proxy to call `map` on another class
* I think this change may make the `LifeCycleConfiguration` a little bit redundant, but would have to see what it looks like first.

Reply:
As per your comment on line 90 in `MessageEventToPathEventMapper`, I think this will be tricky as we'll need to pass the `tableParameters` to the enum method to do this, but then we'll need to pass the keys too, which will make effort this redundant. Hence why this suggestion in the comment above potentially means splitting `ApiaryLifeCycleEvent` into two classes - one which returns an expired and one which returns an orphaned path.

Reply:
100% understand that reasoning. I've split that out into the suggested pattern w/ an abstract pathMapper and the Orphaned/Expired PathMappers extending it to share some utility methods between them.